### ISSUE TITLE
Add missing network configurations with `Client`

### ISSUE THREAD

**Issue by RickyLB** [hiero-ledger/hiero-sdk-rust#845]
## 🆕🐥 First Timers Only

This issue is reserved for people who have never contributed to [Hedera](https://hedera.com) or any open source project in general.
We know that creating a pull request (PR) is a major barrier for new contributors.
The goal of this issue and all other issues labeled by [**'good first issue'**](https://github.com/issues?q=is%3Aopen+is%3Aissue+org%3Ahashgraph+archived%3Afalse+label%3A%22good+first+issue%22+) is to help you make your first contribution to Hedera.

## 👾 Description of the issue

These SDK methods for the Hedera network are critical for configuring client behavior and ensuring robust network communication. The methods allow for setting and checking transport security (TLS), managing retry attempts for nodes with failed gRPC statuses, and controlling the number of nodes involved in each transaction, which impacts both security and network efficiency.

The Rust SDK is missing the following `Client` methods that allow developers to configure the `Network` behavior in [client/mod.rs](https://github.com/hashgraph/hedera-sdk-rust/blob/main/src/client/mod.rs).

``` rust
impl Client {
	///
	/// Set if transport security should be used to connect to consensus nodes.
	///
	pub fn set_transport_security(&self, transport_security: bool) {}

	///
	/// Is tls enabled for consensus nodes.
	///
	pub fn is_transport_security(&self) {}

	///
	/// Max number of times any node in the network can receive a bad gRPC status 
	///
	pub fn get_max_node_attempts(&self) {}

	///
	/// Set the max number of times any node in the network can receive a bad gRPC status before being removed from the network.
	///
	pub fn set_max_node_attempts(&self, max_node_attempts: u32) {}

	///
    /// Set the max amount of nodes that will be chosen per request. By default, the request will use 1/3rd the network
    /// nodes per request.
    ///
	pub fn set_max_nodes_per_transaction(&self, max_nodes_per_transaction: u32) {}
}
```

## Solution 

These methods exist in the Java SDK  ([sdk/Client.java](https://github.com/hashgraph/hedera-sdk-java/blob/main/sdk/src/main/java/com/hedera/hashgraph/sdk/Client.java))

## Implementation 
1. Create logic for each method within `Impl NetworkData`
2. Expose methods to configure network using `Client`

## 📋 Step by step guide to do a contribution

If you have never contributed to an open source project at GitHub, the following step-by-step guide will introduce you to the workflow. More information and concrete samples for shell commands for each step can be found in our [CONTRIBUTING.md](https://github.com/hashgraph/.github/blob/main/CONTRIBUTING.md) file.
A more detailed general documentation of the GitHub PR workflow can be found [here](https://github.com/firstcontributions/first-contributions/blob/master/README.md).

- [ ] **Claim this issue:** Comment below that you are interested in working on the issue
- [ ] **Wait for assignment:** A community member with the given rights will add you as an assignee of the issue
- [ ] **Fork the repository:** You can do that in GitHub (by simply clicking the 'fork' button).
- [ ] **Check out the forked repository**
- [ ] **Create a feature branch** for the issue. We do not have a hard naming definition for branches but it is best practice to prefix the branch name with the issue id.
- [ ] **Solve the issue** in your branch.
- [ ] **Commit your changes:** Here, it is needed to add `sign-off` information to the commit to accept the "Developer Certificate of Origin" (https://developercertificate.org). More details can be found in our [CONTRIBUTING.md](https://github.com/hashgraph/.github/blob/main/CONTRIBUTING.md)
- [ ] **Start a Pull Request (PR)**: We have a pattern for naming pull requests that a GitHub Action checks. We use that pattern to support the creation of automatic release notes.
- [ ] **Check GitHub Actions:** Several GitHub Actions will be triggered automatically for each PR. If a GitHub Action fails and you do not understand the cause of that error do not hesitate to add a comment to the PR and ask the Hedera developer community for support.
- [ ] **Wait for reviews:** Members of the Hedera developer community will review your PR. If a reviewer finds any missing pieces or a problem, he or she will start a discussion with you and describe the next steps for solving the problem.
- [ ] **You did it 🎉:** We will merge the fix in the develop branch. Thanks for being part of the Hedera community as an open-source contributor ❤️

## 🎉 Contribute to Hacktoberfest

Solve this issue as part of the [Hacktoberfest](https://hacktoberfest.digitalocean.com) event and get a chance to receive cool goodies like a T-Shirt. 🎽

## 🤔 Additional Information

If you have any questions, just ask us directly in this issue by adding a comment. You can join our community chat at [Discord](https://hedera.com/discord). A general manual about open-source contributions can be found [here](https://github.com/firstcontributions/first-contributions/blob/master/README.md).

**Comment by shivaram19** [comment]
hello , I'm shivaram , I've been working on the rust lately , could you assign me this issue

**Comment by hendrikebbers** [comment]
@shivaram19 do you still want to work on this issue?

**Comment by shivaram19** [comment]
yes , @hendrikebbers ! I'd still work on this issue . Thank you tracing back !

**Comment by Logan-Pageler** [comment]
With this issue being open for so long with no additional development, could I take a crack at it?

**Comment by Logan-Pageler** [comment]
I made a pr to implement `set_max_nodes_per_transaction`. It seems like there were already implementations for `get_max_node_attempts` (`client.max_node_attempts`) and `set_max_node_attempts`.

Additionally it seems like the rust sdk currently doesnt support tls (see #1160) so I figured thatd be out of the scope of just one pr.
